packages/ledgerstate/output.go

```diff
@@ -86,11 +118,16 @@ func OutputIDFromBase58(base58String string) (outputID OutputID, err error) {
 func OutputIDFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (outputID OutputID, err error) {
 	outputIDBytes, err := marshalUtil.ReadBytes(OutputIDLength)
 	if err != nil {
-		err = xerrors.Errorf("failed to parse OutputID (%v): %w", err, ErrParseBytesFailed)
+		err = xerrors.Errorf("failed to parse OutputID (%v): %w", err, cerrors.ErrParseBytesFailed)
 		return
 	}
 	copy(outputID[:], outputIDBytes)
 
+	if outputID.OutputIndex() >= MaxOutputCount {
+		err = xerrors.Errorf("output index exceeds threshold defined by MaxOutputCount (%d): %w", MaxOutputCount, cerrors.ErrParseBytesFailed)
```

Comment:
I discussed this with angelo and jonas and we decided to not have a separate "ErrSyntacticalValidationFailed" error because these things can also happen through things like bitflips on the transport layer and it essentially makes no sense to differentiate between Syntactical Errors and other Parsing error because any violation of the length or layout of the marshaled bytes is equally syntactically invalid as for example not sorting the outputs accordingly.

So we decided to have 1 generic ErrParseBytesFailed error but instead be a bit more specific in the message.